src/environment.cpp

```diff
@@ -1583,6 +1583,31 @@ u32 ServerEnvironment::addParticleSpawner(float exptime)
 	return id;
 }
 
+u32 ServerEnvironment::addParticleSpawner(float exptime, u16 attached_id)
+{
+	u32 id = addParticleSpawner(exptime);
+	m_particle_spawner_attachments[id] = attached_id;
+	ServerActiveObject *obj = getActiveObject(attached_id);
```

Comment:
no you use it only in the condition before, then you can do this affectation in the condition